torch/csrc/jit/runtime/decomposition_registry.cpp

```diff
@@ -160,6 +160,26 @@ void RegisterDecomposition(
   schema_to_decomposition[&schema] = g;
 }
 
+void run_jit_decomposition(
+    const c10::OperatorHandle& op,
+    torch::jit::Stack* stack) {
+  const auto& schema = op.schema();
+  // TODO: templatize based on op and keep static trace_exec
+  auto* trace_exec = torch::jit::GetDecompositionExecutor(schema);
+  trace_exec->run((*stack));
+  if (stack->back().isTuple()) {
```

Comment:
Why do we unpack tuple results here?

Reply:
Hmm not completely sure. This was from https://github.com/pytorch/functorch/pull/778, maybe @eellison knows?

Reply:
JIT returns a boxed tuple as a single IValue but pytorch core etc wants unboxed, multiple values on stack.